lib/matplotlib/_mathtext.py

```diff
@@ -1947,6 +1947,9 @@ def __init__(self):
         self._expression = p.main
         self._math_expression = p.math
 
+        # To add space to nucleus operators after sub/superscripts
+        self._subsuper_flag = False
```

Comment:
Slightly clearer naming:
```suggestion
        self._in_subscript_or_superscript = False
```